resolversrv/tests/unittests/test_db_query.py

```diff
@@ -8,186 +8,424 @@
+                        ('2013MNRAS.435.1904M', 'DATA',         'SIMBAD',      ['http://$SIMBAD$/simbo.pl?bibcode=2013MNRAS.435.1904M'], ['SIMBAD Objects (30)'], 30),
+                        ('2013MNRAS.435.1904M', 'DATA',         'XMM',         ['http://nxsa.esac.esa.int/nxsa-web/#obsid=0097820101'], ['XMM-Newton Observation Number 0097820101'], 1),
+                        ('2017MNRAS.467.3556B', 'PRESENTATION', '',            ['http://www.astro.lu.se/~alexey/animations.html'], [''], 0),
+                        ('1943RvMP...15....1C', 'INSPIRE',      '',            ['http://inspirehep.net/search?p=find+j+RMPHA,15,1'], [''], 0),
+                        ('1971ATsir.615....4D', 'ASSOCIATED',   '',            ['1971ATsir.615....4D', '1974Afz....10..315D', '1971ATsir.621....7D', '1976Afz....12..665D', '1971ATsir.624....1D', '1983Afz....19..229D', '1983Ap.....19..134D', '1973ATsir.759....6D', '1984Afz....20..525D', '1984Ap.....20..290D', '1974ATsir.809....1D', '1974ATsir.809....2D', '1974ATsir.837....2D'], ['Part  1', 'Part  2', 'Part  3', 'Part  4', 'Part  5', 'Part  6', 'Part  7', 'Part  8', 'Part  9', 'Part 10', 'Part 11', 'Part 12', 'Part 13'], 0)
                     ]
 
+        record_list_msg = DataLinksRecordList()
+        record_list_msg.status = 2 # new
         for record in stub_data:
-            data_link = DataLinks(bibcode=record[0],
-                                  link_type=record[1],
-                                  link_sub_type=record[2],
-                                  url=record[3],
-                                  title=record[4],
-                                  item_count=record[5])
-            self.app.db.session.add(data_link)
-        self.app.db.session.commit()
+            datalinks_record = {'bibcode': record[0],
+                                'data_links_rows': [{'link_type': record[1], 'link_sub_type': record[2],
+                                                     'url': record[3], 'title': record[4],
+                                                     'item_count': record[5]}]}
+            record_list_msg.datalinks_records.add(**datalinks_record)
+        status, text = add_records(record_list_msg)
+        self.assertEqual(status, True)
+        self.assertEqual(text, 'updated db with new data successfully')
+
+
+    def test_add_records_no_data(self):
+        """
+        return False and text explanation when an empty DataLinksRecordList is passed to add_records
+        :return:
+        """
+        status, text = add_records(DataLinksRecordList())
+        self.assertEqual(status, False)
+        self.assertEqual(text, 'unable to extract data from protobuf structure')
+
+
+    def test_add_records_wrong_message_structure(self):
+        """
+        return False and text explanation when an structure other than DataLinksRecordList is passed to add_records
+        :return:
+        """
+        status, text = add_records(DataLinksRecord())
```

Comment:
consider throwing an exception

Reply:
Thank you for noticing this. turned out that after talking with Steve and deciding that we are only supporting DataLinksRecordList, I do not need this test anymore.